vpr/src/draw/search_bar.cpp

```diff
@@ -339,34 +420,135 @@ void warning_dialog_box(const char* message) {
+    GtkEntryCompletion* completion = GTK_ENTRY_COMPLETION(app->get_object("Completion"));
+    GtkTreeModel* netNames = GTK_TREE_MODEL(app->get_object("NetNames"));
+    GtkTreeModel* blockNames = GTK_TREE_MODEL(app->get_object("BlockNames"));
     //Ensuring a valid type was selected
     if (!type) return;
     if (type[0] == '\0') return;
     std::string searchType(type);
-    //Setting active completion model to blockCompleter if type selected is block Name
+
+    /*
+     * If search type is name, connecting search bar to completion,
+     * and connecting completion to the appropriate model (blocks or nets)
+     * Additionally, visibility of key length setter is toggled by these changes
+     */
     if (searchType == "Block Name") {
-        GtkEntryCompletion* blockCompleter = GTK_ENTRY_COMPLETION(app->get_object("BlockNameCompleter"));
-        gtk_entry_set_completion(searchBar, blockCompleter);
-    } else { //If not, setting to null
+        gtk_entry_completion_set_model(completion, blockNames);
+    } else if (searchType == "Net Name") {
+        gtk_entry_completion_set_model(completion, netNames);
+    } else { //setting to null if option does not require auto-complete
+        gtk_entry_completion_set_model(completion, NULL);
         gtk_entry_set_completion(searchBar, nullptr);
     }
 }
 
 /**
- * @brief loads block names into gtk list store item used for completion
+ * @brief A non-default matching function. As opposed to simply searching for a prefix(default),
+ * searches string for presence of a substring. Ignores cases.
  * 
- * @param app ezgl application
+ * @param completer the GtkEntryCompletion being used
+ * @param key a normalized and case-folded key representing the text
+ * @param iter GtkTreeIter pointing at the current entry being compared
+ * @param user_data null
+ * @return gboolean 
  */
-void load_block_names(ezgl::application* app) {
-    auto blockStorage = GTK_LIST_STORE(app->get_object("BlockNames"));
-    auto& cluster_ctx = g_vpr_ctx.clustering();
-    GtkTreeIter iter;
-    //Getting and storing all block names
-    for (ClusterBlockId id : cluster_ctx.clb_nlist.blocks()) {
-        gtk_list_store_append(blockStorage, &iter);
-        gtk_list_store_set(blockStorage, &iter,
-                           0, (cluster_ctx.clb_nlist.block_name(id)).c_str(), -1);
+gboolean customMatchingFunction(
+    GtkEntryCompletion* completer,
+    const gchar* key,
+    GtkTreeIter* iter,
+    gpointer /*user data*/
+) {
+    GtkTreeModel* model = gtk_entry_completion_get_model(completer);
+    const gchar* text;
+    gtk_tree_model_get(model, iter, 0, &text, -1);
+    //Removing case information
+    g_utf8_casefold(text, -1);
+    g_utf8_normalize(text, -1, G_NORMALIZE_DEFAULT);
+    std::string cppText(text);
+    //If substring not found, returning false;
+    return (cppText.find(key, 0) != std::string::npos);
+}
+
+/**
+ * @brief Creates a GdkEvent that simulates user pressing key "key"
+ * 
```

Comment:
Might be useful to say what this is currently used for.
Also, I think it'll be better if the function level Doxygen comments move to the .h file.